app/domain/authentication/authn_gcp/authenticator.rb

```diff
@@ -0,0 +1,44 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGcp
+
+    Authenticator = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       [:authenticator_input]
+    ) do
+
+      extend Forwardable
+      def_delegators :@authenticator_input, :account, :credentials, :username
+
+      def call
+        # TODO: validate resource restrictions
+        # Note: 'credentials' is now the decoded token. you can access its
+        # fields with an argument reader (e.g credentials.project_id)
+      end
+
+      private
```

Comment:
Does this `private` has a meaning?

Reply:
not at the moment but i'm assuming that private methods will be added later. i wouldn't bother with this in this PR